src/app/pages/rpc-stx-transfer-stx/use-rpc-stx-transfer-stx.ts

```diff
@@ -0,0 +1,43 @@
+import { useMemo } from 'react';
+
+import { createRequestEncoder, stxTransferStx } from '@leather.io/rpc';
+import { type StacksUnsignedTokenTransferOptions, TransactionTypes } from '@leather.io/stacks';
+import { createMoney, isDefined } from '@leather.io/utils';
+
+import { initialSearchParams } from '@app/common/initial-search-params';
+import { useBreakOnNonCompliantEntity } from '@app/query/common/compliance-checker/compliance-checker.query';
+import { useCurrentStacksAccount } from '@app/store/accounts/blockchain/stacks/stacks-account.hooks';
+import { useCurrentStacksNetworkState } from '@app/store/networks/networks.hooks';
+
+const { decode } = createRequestEncoder(stxTransferStx.request);
+
+function useStxTransferStxRequest() {
+  const request = initialSearchParams.get('rpcRequest');
+  if (!request) throw new Error('Missing rpcRequest');
+  return { request: decode(request) };
+}
+
+export function useRpcStxTransferStxTxOptions(): StacksUnsignedTokenTransferOptions {
+  const { request } = useStxTransferStxRequest();
+  const account = useCurrentStacksAccount();
+  const network = useCurrentStacksNetworkState();
+
+  useBreakOnNonCompliantEntity(
+    [request.params.address ?? account?.address, request.params.recipient].filter(isDefined)
+  );
+
+  return useMemo(
+    () => ({
+      txType: TransactionTypes.StxTokenTransfer,
+      amount: createMoney(request.params.amount, 'STX'),
+      fee: createMoney(request.params.fee ?? 0, 'STX'),
+      memo: request.params.memo ?? '',
+      recipient: request.params.recipient,
+      network,
+      nonce: request.params.nonce ?? 0,
+      publicKey: account?.stxPublicKey || '',
+      sponsored: request.params.sponsored ?? false,
+    }),
+    [account, network, request]
```

Comment:
Just wondering if this should better be a function, where rather than say, defaulting to zero fee, you handle an error case?

Reply:
I don't think it matters here to handle an error state with conditional params from the request? I believe we use defaults just to get past type errors for missing types in the required options to actually calculate the fee estimations for the unsigned tx. We then later use the fee/nonce editor fee and nonce before signing and broadcasting. I can prob make the a function though and avoid using the hook.